### PR TITLE
[GEN] Use OCL builtin for non-tf32 dpas

### PR DESCRIPTION
**Pull request by whitneywhtsang**
OCL builtins used in this PR are defined in https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroup_matrix_multiply_accumulate.html

### CODE REVIEW

**Review by chengjunlu — APPROVED** [review]

**Review by Dewei-Wang-sh — COMMENTED** [review]
LGTM.
just one question, did this verified in a end2end gemm case?

**Comment by chengjunlu** [comment]
> LGTM. just one question, did this verified in a end2end gemm case?

Good point. @whitneywhtsang Can you run the end2end test by hard code the threads_per_warp=16 like this?
https://github.com/intel/intel-xpu-backend-for-triton/blob/25f9d248ccb3bfafee9ffac69b0a259b045837be/third_party/intel/backend/compiler.py#L97

It changes the default threads_per_warp to 16 and chose an optimized num_warps correspondingly.

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 1]
```diff
@@ -158,6 +172,18 @@ static LLVM::CallOp createGenISADPAS(TritonGEN::MatrixDPASOp op,
   if (bOrigTy != bTy)
     b = rewriter.create<LLVM::BitcastOp>(loc, bTy, b);
 
+  if (precisionA != TritonGEN::PrecisionType::TF32) {
```
Comment (etiotto) [inline third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:175]: Add a FIXME: we will want to use the OpenCL API also for the remaining TF32 case.

**Review by etiotto — APPROVED** [review]

**Comment by whitneywhtsang** [comment]
> @whitneywhtsang Can you run the end2end test/unit/operators/test_matmul.py by hard code the threads_per_warp=16 like this?

Verified that there are no regressions, but not all test_matmul.py variances pass.

**Comment by chengjunlu** [comment]
> > @whitneywhtsang Can you run the end2end test/unit/operators/test_matmul.py by hard code the threads_per_warp=16 like this?
> 
> Verified that there are no regressions, but not all test_matmul.py variances pass with threads_per_warp=16.

Thanks for the confirmation.
Some failures are expected and tracked in the https://github.com/intel/intel-xpu-backend-for-triton/issues/718.

**Pull request merged**